library/core/src/mem/maybe_uninit.rs

```diff
@@ -246,6 +246,14 @@ impl<T> MaybeUninit<T> {
     /// Note that dropping a `MaybeUninit<T>` will never call `T`'s drop code.
     /// It is your responsibility to make sure `T` gets dropped if it got initialized.
     ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::mem::MaybeUninit;
+    ///
+    /// let v = MaybeUninit::new(vec![42]);
```

Comment:
```suggestion
    /// let v: MaybeUninit<Vec<u8>> = MaybeUninit::new(vec![42]);
```
Add a type for clarification.